src/symbol/placement.js

```diff
@@ -362,16 +368,13 @@ class Placement {
     }
 
     hasTransitions(now: number) {
-        return this.symbolFadeChange(now) < 1 || this.stale;
+        return this.stale ||
+            now - this.lastPlacementChangeTime < this.fadeDuration;
     }
 
     stillRecent(now: number) {
-        return this.recentUntil > now;
-    }
-
-    setRecent(now: number, stale: boolean) {
```

Comment:
Love how this change gets rid of the need for `setRecent/recentUntil` -- no net increase in internal state!